tst/testrandom.g

```diff
@@ -59,35 +77,47 @@ randomTestForSizeOneCollection := function(collection, method)
 
     val := Representative(collection);
 
-    Init(GlobalMersenneTwister, 6);
-    intlist1 := List([1..1000], x -> Random([1..10]));
+    Init(globalobj, 6);
+    intlist1 := List([1..10], x -> gmethod([1..1000]));
 
-    for i in [1..1000] do
-        if method(collection) <> val then
+    for i in [1..100] do
+        if gmethod(collection) <> val then
             Print("Random returned something outside collection :", collection, ":", val);
         fi;
     od;
 
-    for i in [1..1000] do
-        if method(GlobalMersenneTwister, collection) <> val then
+    for i in [1..100] do
+        if method(globalobj, collection) <> val then
             Print("Random returned something outside collection :", collection, ":", val);
         fi;
     od;
 
-    localgen := RandomSource(IsMersenneTwister, 6);
+    localgen := RandomSource(filter, 6);
 
-    Init(GlobalMersenneTwister, 6);
-    for i in [1..1000] do
+    Init(globalobj, 6);
+    for i in [1..100] do
         if method(localgen, collection) <> val then
             Print("Random returned something outside collection :", collection, ":", val);
         fi;
     od;
 
-    # The previous loop should not have affected GlobalMersenneTwister,
+    # The previous loop should not have affected globalobj,
     # so this should be the same as intlist1
-    intlist2 := List([1..1000], x -> Random([1..10]));
+    intlist2 := List([1..10], x -> gmethod([1..1000]));
 
     if intlist1 <> intlist2 then
         Print("Random read from local gen affected global gen: ", collection);
     fi;
 end;;
+
+randomTestForSizeOneCollection := function(collection, method, checkin...)
```

Comment:
BTW, I wonder: is there a strong reason to have `randomTestForSizeOneCollection` be a separate function? Couldn't `randomTest` choose between calling `randomTestInner` and `randomTestForSizeOneCollectionInner` based on the collection size? (Even better, it could use `IsTrivial` on the collection, which means "size = 1" but may work even on collections which don't know their precise size)

Reply:
Yes, we should do that.